src/Compilers/CSharp/Portable/Symbols/ExtraAnnotations.cs

```diff
@@ -18,6 +18,8 @@ internal enum AttributeAnnotations
         NotNullWhenTrue = 1 << 0,
         NotNullWhenFalse = 1 << 1,
         EnsuresNotNull = NotNullWhenTrue | NotNullWhenFalse,
+        EnsuresTrue = 1 << 2,
+        EnsuresFalse = 1 << 3,
     }
```

Comment:
`AttributeAnnotations` is not specific to the `ExtraAnnotations` class. Consider moving to a separate file.

Also, consider renaming, perhaps to `FlowAnalysisAnnotations`, to match the `Symbol` properties.